src/LibraryUtilities.ts

```diff
@@ -249,3 +255,32 @@ export function buildLibraryItemsFromLayoutSpecs(loadedTypes: any, layoutSpecs:
     }
     return convertToLibraryTree(typeListNodes, layoutElements);
 }
+
+function setItemVisible(item: ItemData) {
+    item.visible = true;
+    item.childItems.forEach((childItem) => setItemVisible(childItem));
+}
+
+export function search(text: string, item: ItemData) {
+    let index = item.text.toLowerCase().indexOf(text);
```

Comment:
I think once `index` is set and it is not `-1`, then we can make the item visible and return, right? Otherwise the `for` loop below will overwrite its value.